### ISSUE TITLE
Address-controller is restarted with 'Error syncing pod' event and no additional info in log

### ISSUE THREAD

**Issue by tkratky** [EnMasseProject/enmasse#385]
### description: 
- when 2 addresses are deployed into default address space (1 message is send to first of them) and one by one are removed then address controller is restarted with following events:

2:06:54 PM | Warning | Unhealthy | Liveness  probe failed: Get http://172.17.0.9:8080/v1/health: net/http: request  canceled (Client.Timeout exceeded while awaiting headers)    10 times in the last 31 minutes
-- | -- | -- | --
2:00:14 PM | Warning | Unhealthy | Readiness  probe failed: Get http://172.17.0.9:8080/v1/health: net/http: request  canceled (Client.Timeout exceeded while awaiting headers)    9 times in the last 31 minutes
1:53:07 PM | Warning | Back off | Back-off restarting failed container
1:53:07 PM | Warning | Failed sync | Error syncing pod

### prerequisites:
- enmasse running in multitenant mode: `sh ./templates/install/deploy-openshift.sh -a "none standard" -o "multitenant"`

### steps to reproduce:
1. execute already existing test: `make MULTITENANT=true SYSTEMTEST_ARGS='standard.QueueTest#testCreateDeleteQueue' systemtests`
https://github.com/EnMasseProject/enmasse/blob/fix_delete_address/systemtests/src/test/java/io/enmasse/systemtest/standard/QueueTest.java#L118
2. output:
Address-controller is restarted and following html is returned
```

  <body>
    <div>
      <h1>Application is not available</h1>
      <p>The application is currently not serving requests at this endpoint. It may not have been started or is still starting.</p>

      <div class="alert alert-info">
        <p class="info">
          Possible reasons you are seeing this page:
        </p>
        <ul>
          <li>
            <strong>The host doesn't exist.</strong>
            Make sure the hostname was typed correctly and that a route matching this hostname exists.
          </li>
          <li>
            <strong>The host exists, but doesn't have a matching path.</strong>
            Check if the URL path was typed correctly and that the route was created using the desired path.
          </li>
          <li>
            <strong>Route and path matches, but all pods are down.</strong>
            Make sure that the resources exposed by this route (pods, services, deployment configs, etc) have at least one pod running.
          </li>
        </ul>
      </div>
    </div>
  </body>
</html>


```
With connection error:
```
262164 [vert.x-eventloop-thread-0] WARN test - Error when getting addresses
io.vertx.core.VertxException: Connection was closed
	at io.vertx.core.http.impl.ClientConnection.handleClosed(ClientConnection.java:406)
	at io.vertx.core.impl.ContextImpl.lambda$wrapTask$2(ContextImpl.java:342)
	at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:200)
	at io.vertx.core.net.impl.VertxHandler.channelInactive(VertxHandler.java:134)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:224)
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelInactive(CombinedChannelDuplexHandler.java:420)
	at io.netty.handler.codec.ByteToMessageDecoder.channelInputClosed(ByteToMessageDecoder.java:377)
	at io.netty.handler.codec.ByteToMessageDecoder.channelInactive(ByteToMessageDecoder.java:342)
	at io.netty.handler.codec.http.HttpClientCodec$Decoder.channelInactive(HttpClientCodec.java:281)
	at io.netty.channel.CombinedChannelDuplexHandler.channelInactive(CombinedChannelDuplexHandler.java:223)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:224)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1354)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:245)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:231)
	at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:917)
	at io.netty.channel.AbstractChannel$AbstractUnsafe$7.run(AbstractChannel.java:763)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:403)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:463)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at java.lang.Thread.run(Thread.java:748)

```

**Comment by lulf** [comment]
Thanks for the great bug description @tkratky , thanks to that I was able to reproduce quite easily (although not every time). The reason it crashes is due to OOM: 

address-controller-3442122358-zqtxt    0/1       OOMKilled   0          15m

I will bump the limit for now, though I'll do some profiling as well to determine whats using memory.

**Comment by tkratky** [comment]
issue with OOM happened again

**Comment by lulf** [comment]
This is fixed for real in #448

**Comment by lulf** [comment]
See https://bugs.openjdk.java.net/browse/JDK-8187822
